Modernised Find and Replace UI

jncn opened the pull request:
Modernised the Find and Replace UI, and some additional clean up:
- The Find and Replace UI has now a more space efficient lay out, I think without compromises
- Fixes a bug where the Find All window would clip text until resizing its window
- Smaller lay out improvements for the Find All window
- Removed HUD buttons style from DocumentProxy
- Removed old assets (even some nice Pather-style buttons 💔)

And of course some other commits sneaked in again, not sure why this is happening.

jncn committed:
Replaced URLBubbleWindow with OpenURLViewController and NSPopover

jncn committed:
Removed all instanced of URLBubbleWindow

jncn committed:
Updated File's Owner class name in IB

jncn committed:
Merge branch 'develop' of https://github.com/subethaedit/SubEthaEdit into develop

# Conflicts:
#	SubEthaEdit-Mac/Source/OpenURLViewController.h
#	SubEthaEdit-Mac/Source/OpenURLViewController.m
#	SubEthaEdit-Mac/SubEthaEdit.xcodeproj/project.pbxproj
#	SubEthaEdit-Mac/de.lproj/OpenURLViewController.xib

jncn committed:
Fixed some lay out issues with the Web Preview

jncn committed:
Merge branch 'develop' of https://github.com/subethaedit/SubEthaEdit into develop

jncn committed:
Removed unused assets

jncn committed:
Removed HUD button from DocumentProxy

jncn committed:
- Fixed issue with clipped results in FindAllController
- Cleaned up the appearance a bit of FindAllController

jncn committed:
Modernised `SEEFindAndReplaceViewController`

jncn committed:
German localisation additions for Find and Replace

jncn committed:
Added tooltips to the next/prev buttons

jncn committed:
Fixed IB warnings about clipping

monkeydom committed:
Merge branch 'develop' into improvement/findandreplace-ui

# Conflicts:
#	SubEthaEdit-Mac/Source/SEEWebPreviewViewController.m

monkeydom committed:
[META] add line to changelog

monkeydom commented:
![Pasted_Image_12_12_20__19_18](https://user-images.githubusercontent.com/5106/101991735-e2ae9380-3cae-11eb-9798-591525673de2.jpg)


Addding a after and before comparison image. Overall I do think this is a positive change, but I'll need some getting used to it. The gear dropdown is a little bit cryptic, and when regex is off the actual gear does not align nicely either. `(.*)` doesn't say too much regex to me, and bracing evertyhing just makes `\1` equal `\0` so hmmm. 

So to gather some open issues:
* the inline result text overlaps and looks broken if the regex is actually filling the line
* the gear is not aligned properly
* the regex mode of the gear is not expressive enough

monkeydom commented:
![Pasted_Image_12_12_20__19_38](https://user-images.githubusercontent.com/5106/101992191-b0eafc00-3cb1-11eb-9366-8741dde006bf.jpg)
Example of a replace all result text overwriting the long regex.

monkeydom commented:
![Pasted_Image_12_12_20__19_43](https://user-images.githubusercontent.com/5106/101992272-48e8e580-3cb2-11eb-8630-44af42131ae2.jpg)
German localizations seems to also show "1 replaced" - might be bug existing b ut should be fixed either wy.

monkeydom commented on SubEthaEdit-Mac/Base.lproj/Localizable.strings:
```diff
@@ -23,13 +23,13 @@
 "%d pending" = "%d pending";
 
 /* Number of replaced strings */
-"%d replaced." = "%d replaced.";
```
here is the localization culprit. if you change the keys you need to touch the german loclziation too

jncn replied:
Missed that one, good catch!

monkeydom replied:
maybe it isn't? seems to be fixed in the german localization. maybe the problem is somewhere else.

monkeydom commented on SubEthaEdit-Mac/Source/SEEFindAndReplaceState.m:
```diff
@@ -276,11 +276,11 @@ - (void)takeValuesFromDictionaryRepresentation:(NSDictionary *)aDictionaryRepres
 
 - (NSString *)menuTitleDescription {
 	NSMutableArray *components = [NSMutableArray new];
-	[components addObject:[NSString stringWithFormat:@"›%@‹", self.findString]];
```
Don't know f I see the removal of the angle braces an improvement.

jncn replied:
Fair enough! It confused me when I first saw it in that menu, I thought it was a glitch. I've never seen these in use before. 

I looked around a bit and found out other apps aren't using any braces in their recents. Which doesn't mean SEE should do the same, of course 😉

monkeydom replied:
I do agree all of this is quite cryptic though. If you like it clearly better this way I might get convinced as less often is more.

jncn commented:
Alright! Thanks for the review. I'll do a proposal for the dropdown and fix the label.

monkeydom committed:
[META] adjusted asset filename

monkeydom reviewed (changes requested):
Requesting some changes as in the discussion. But looks good in general.

jncn committed:
- Fixed localisation issue
- Removed some unused IBOutlets

jncn committed:
- Fixed overlapping status text
- Moved the magnifying glass button back to the left
- Changed the RegEx icon

jncn commented:
I played around with the lay out a bit again, I think we're nearly there:
- Fixed the localisation issue
- Fixed the overlapping status text 
- Moved the options button back to the left
- Changed the RegEx icon to `*`

One open question is backwards compatibility. I find myself writing more and more checks for macOS 10.14+ (because quite a few UI-API's changed with the introduction of Dark Mode). Would it be a good idea to drop 10.13 and older for 5.2?
Because of this, I haven't tackled the issue of tinting the button when a search with a scope is performed (the shadow didn't look good).

Feedback welcome! 

Here are already some screenshots:

<img width="717" alt="Screenshot 2020-12-13 at 01 03 39" src="https://user-images.githubusercontent.com/995815/101999612-16a1ad00-3cdf-11eb-8f7a-171ee66a71cb.png">

<img width="717" alt="Screenshot 2020-12-13 at 00 46 46" src="https://user-images.githubusercontent.com/995815/101999505-210f7700-3cde-11eb-9ee3-f275b417794e.png">

monkeydom commented:
I'm open to requiring 10.14 aka mojave. It seems like a solid progression. It is also easier for me to actually make sure it works, as my 10.13 machine is probably not for much longer anyways.

And if it turns out that we do have a good case for macOS 11 only because of major style questions, I'm open to that as well, as we do have a solid release for the lower versions and I do think the macOS 11 style really begs for bigger changes that probably feel a little alien to 10.15/10.14 anyways.

monkeydom reviewed (commented):
Looks good to me. Need to do the manual diff, build an test dance before actually approving.

monkeydom committed:
[META] just some simple whitespace and style adjustments

monkeydom reviewed (approved):
Looks very good to me, thanks!
